### PR TITLE
[SYCL] Fix SYCL includes in SYCLDeviceManager

### PR DESCRIPTION
**Pull request by stephan-rohr**
The includes 'CL/sycl/backend.hpp' and 'CL/sycl/backend/level_zero.hpp' are already included in 'CL/sycl.hpp' in 'SYCLDeviceManager.h' such that we can remove that here. 
Also, I could not find any calls to level_zero api such that I suggest to remove this include.

Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

This commit removes non needed includes in SYCLDeviceManager.cpp to improve code quality. 

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

- Ubuntu 20.04

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by prckent** [comment]
Thanks Stephan.

We don't have SYCL in our CI yet, so I'll wait for confirmation from someone familiar with it. However this Looks Good To Me.

**Review by PDoakORNL — APPROVED** [review]
This LGTM but I think @ye-luo should also sign off.

File: src/Platforms/SYCL/SYCLDeviceManager.cpp [hunk 1]
```diff
@@ -21,9 +21,6 @@
 #include "determineDefaultDeviceNum.h"
 #if defined(_OPENMP)
 #include <omp.h>
-#include <CL/sycl/backend.hpp>
-#include <level_zero/ze_api.h>
```
[the remark this answers is not in the document]
Comment (stephan-rohr) [reply src/Platforms/SYCL/SYCLDeviceManager.cpp:25]: I don't have level_zero package installed on my system causing a build failure. I checked for calls to level_zero but did not find any such that I concluded we can remove this include.

I agree that the 'Fix' label may be not be needed. We can remove this.

[the remark this answers is not in the document]
Comment (ye-luo) [reply src/Platforms/SYCL/SYCLDeviceManager.cpp:25]: It is likely that the latest SYCL libraries avoids the dependency on L0 header files. I never wanted these include lines and it is time to remove them. What is your compiler version? `icpx --version` first line

[the remark this answers is not in the document]
Comment (stephan-rohr) [reply src/Platforms/SYCL/SYCLDeviceManager.cpp:25]: I validated with version
Intel(R) oneAPI DPC++/C++ Compiler 2022.1.0 (2022.1.0.20220316)

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**